File: apps/ios/ScribbleFit/ScribbleFit/Features/Analytics/Domain/UseCases/AnalyzeWorkoutsUseCase.swift
import Foundation

/**
 * UseCase to trigger a background analysis refresh.
 */
public final class AnalyzeWorkoutsUseCase {
    private let repository: AnalysisRepository
    private let engine: AnalysisEngine
    
    public init(repository: AnalysisRepository, engine: AnalysisEngine) {
        self.repository = repository
        self.engine = engine
    }
    
    public func refreshHomeSuggestion(context: String) async throws {
        let suggestion = try await engine.generateSuggestion(context: context)
        try await repository.saveHomeSuggestion(suggestion)
    }
    
    public func refreshSummary(period: SummaryPeriod, workoutData: String) async throws {
        let summary = try await engine.generateSummary(period: period, workoutData: workoutData)
        try await repository.saveSummary(summary)
    }
    
    public func refreshExerciseInsight(exerciseId: String, exerciseName: String, historyData: String) async throws {
        let insight = try await engine.generateExerciseInsight(exerciseName: exerciseName, historyData: historyData)
        // Ensure exerciseId matches local database ID
        let normalizedInsight = ExerciseInsight(
            exerciseId: exerciseId,
            estimated1RM: insight.estimated1RM,
            prDetected: insight.prDetected,
            trendDirection: insight.trendDirection,
            breakdownText: insight.breakdownText,
            timestamp: insight.timestamp
        )
        try await repository.saveExerciseInsight(normalizedInsight)
    }
}

File: apps/android/feature/analytics/domain/src/main/java/com/scribblefit/feature/analytics/domain/usecase/AnalyzeWorkoutsUseCase.kt
package com.scribblefit.feature.analytics.domain.usecase

import com.scribblefit.feature.ai.domain.engine.AnalysisEngine
import com.scribblefit.feature.ai.domain.model.SummaryPeriod
import com.scribblefit.feature.analytics.domain.repository.AnalysisRepository

class AnalyzeWorkoutsUseCase(
    private val engine: AnalysisEngine,
    private val repository: AnalysisRepository
) {
    suspend fun refreshHomeSuggestion(context: String) {
        engine.generateSuggestion(context).onSuccess { repository.saveHomeSuggestion(it) }
    }

    suspend fun refreshSummary(period: SummaryPeriod, workoutData: String) {
        engine.generateSummary(period, workoutData).onSuccess { repository.saveSummary(it) }
    }

    suspend fun refreshExerciseInsight(exerciseId: String, exerciseName: String, historyData: String) {
        engine.generateExerciseInsight(exerciseName, historyData).onSuccess { insight ->
            repository.saveExerciseInsight(insight.copy(exerciseId = exerciseId))
        }
    }
}
